pkg/supervisor/supervisor_test.go

```diff
@@ -99,20 +99,26 @@ func TestSupervisorStart(t *testing.T) {
 }
 
 func TestGetEnv(t *testing.T) {
-	// backup environment vars
+	// backup environment vars, and restore them when test finishes
 	oldEnv := os.Environ()
+	t.Cleanup(func() {
+		for _, e := range oldEnv {
+			kv := strings.SplitN(e, "=", 2)
+			assert.NoError(t, os.Setenv(kv[0], kv[1]))
```

Comment:
I think the main reason why this test doesn't use `t.Setenv()` is because it wants to ensure a clean environment. So that any environment variables that have been set outside of the test cannot influence its outcome.

We still need to reset/restore the env ourselves, so we didn't win anything here by using the new functionality.

A way to get rid of the OS env in testing would be to pass in the env map as a function parameter to `getEnv()`.

Reply:
> A way to get rid of the OS env in testing would be to pass in the env map as a function parameter to getEnv().

Agreed, that would also make possible in the future to use `t.Parallel`. We aren't using right now and using `t.Parallel` is problematic, but could be somehting that we want in the future.